libs/s25main/desktops/dskSelectMap.cpp

```diff
@@ -450,3 +533,58 @@ void dskSelectMap::FillTable(const std::vector<bfs::path>& files)
         table->AddRow({name, author, players, landscapeNames[header.getGfxSet()], size, filePath.string()});
     }
 }
+
+void dskSelectMap::FillCampaignsTable(const std::vector<boost::filesystem::path>& folders)
+{
+    auto* table = GetCtrl<ctrlTable>(14);
+    for(const bfs::path& folder : folders)
+    {
+        if(helpers::contains(brokenCampaignPaths, folder))
+            continue;
+
+        auto const campaignIniFile = folder / "campaign.ini";
+        CampaignSettings campaignSettings(campaignIniFile.string());
+        if(!campaignSettings.Load())
+        {
+            LOG.write(_("Failed to load campaign %1%.\n")) % folder;
+            brokenCampaignPaths.insert(folder);
+            continue;
+        }
+
+        auto campaignMapFolder = bfs::path(folder);
+        if(campaignSettings.campaignDescription.name == "Roman")
+            campaignMapFolder = RTTRCONFIG.ExpandPath(s25::folders::mapsCampaign);
+        else if(campaignSettings.campaignDescription.name == "Continent")
+            campaignMapFolder = RTTRCONFIG.ExpandPath(s25::folders::mapsContinents);
+
+        for(auto const& map : campaignSettings.missions.mapNames)
+        {
+            auto const mapPath = campaignMapFolder / map;
+            auto const luaFilepath = (folder / map).replace_extension("lua");
+            if(!bfs::exists(mapPath))
+            {
+                LOG.write(_("Campaign map %1% does not exist.\n")) % mapPath;
+                brokenCampaignPaths.insert(folder);
+                continue;
+            }
+            if(!bfs::exists(luaFilepath))
+            {
+                LOG.write(_("Campaign map lua file %1% does not exist.\n")) % luaFilepath;
+                brokenCampaignPaths.insert(folder);
+                continue;
+            }
+
+            libsiedler2::Archiv map;
+            if(int ec = libsiedler2::loader::LoadMAP(mapPath, map, true))
+            {
+                LOG.write(_("Failed to load map %1%: %2%\n")) % mapPath % libsiedler2::getErrorString(ec);
+                brokenMapPaths.insert(folder);
+                continue;
+            }
+        }
+
+        table->AddRow({s25util::ansiToUTF8(campaignSettings.campaignDescription.name),
+                       s25util::ansiToUTF8(campaignSettings.campaignDescription.shortDescription),
+                       s25util::ansiToUTF8(campaignSettings.campaignDescription.author), folder.string()});
```

Comment:
I'd rather expect those to be UTF8 already. Could be ensured or at least checked on loading the settings.